src/txn/txn_rollback_to_stable.c

```diff
@@ -1537,16 +1547,18 @@ __rollback_to_stable_btree_apply(WT_SESSION_IMPL *session)
          * Truncate history store entries for the non-timestamped table.
          * Exceptions:
          * 1. Modified tree - Scenarios where the tree is never checkpointed lead to zero
-         * durable timestamp even they are timestamped tables. Until we have a special indication
-         * of letting to know the table type other than checking checkpointed durable timestamp
-         * to WT_TS_NONE, We need this exception.
+         * durable timestamp even they are timestamped tables. Until we have a special
+         * indication of letting to know the table type other than checking checkpointed durable
+         * timestamp to WT_TS_NONE, We need this exception.
          * 2. In-memory database - In this scenario, there is no history store to truncate.
          */
-        if (!S2BT(session)->modified && max_durable_ts == WT_TS_NONE &&
-          !F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+
+        if (S2BT_SAFE(session) != NULL && (!dhandle_allocated || !S2BT(session)->modified) &&
```

Comment:
I understood the need for this check `S2BT_SAFE(session)` as we are reading the btree id from the dhandle. Do we have any btree id that is stored as part of the metadata that can be used instead of accessing it from the dhandle?

Reply:
Fixed